humanoid_league_speaker/humanoid_league_speaker/speaker.py

```diff
@@ -19,162 +23,108 @@ def speak(text, publisher, priority=20, speaking_active=True):
         publisher.publish(msg)
 
 
-class Speaker:
-    """ Uses espeak to say messages from the speak topic.
+class Speaker(Node):
+    """ 
+    Uses tts to say messages from the speak topic.
     """
 
-    # todo make also a service for demo proposes, which is blocking while talking
+    def __init__(self) -> None:
+        """ Initializes the node and the parameters. """
+        super().__init__("tts_speaker")
 
-    def __init__(self):
-        rclpy.init(args=None)
-        self.node = rclpy.create_node("speaker")
-        self.node.get_logger().info("Starting speaker")
-
-        # --- Class Variables ---
-        self.low_prio_queue = []
-        self.mid_prio_queue = []
-        self.high_prio_queue = []
-        # if you want to change standard startup, do it in the config yaml
+        # Class Variables
+        self.prio_queue: List[Tuple[str, int]] = []
         self.speak_enabled = None
         self.print_say = None
         self.message_level = None
-        self.amplitude = None
-        self.node.declare_parameter("print", True)
-        self.node.declare_parameter("talk", True)
-        self.node.declare_parameter("msg_level", 0)
-        self.node.declare_parameter("amplitude", 7)
-        self.print_say = self.node.get_parameter("print").get_parameter_value().bool_value
-        self.speak_enabled = self.node.get_parameter("talk").get_parameter_value().bool_value
-        self.message_level = self.node.get_parameter("msg_level").get_parameter_value().integer_value
-        self.amplitude = self.node.get_parameter("amplitude").get_parameter_value().integer_value
-        self.node.add_on_set_parameters_callback(self.on_set_parameters)
-
-        self.female_robots = ["donna", "amy"]
-
-        # --- Initialize Topics ---
-        self.node.create_subscription(Audio, "speak", self.speak_cb, 10)
-
-        # --- Start loop ---
-        self.run_speaker()
-
-    def on_set_parameters(self, parameters: [Parameter]):
+        
+        # Initialize Parameters
+        self.declare_parameter("print", True)
+        self.declare_parameter("talk", True)
+        self.declare_parameter("msg_level", 0)
+        self.print_say: bool = self.get_parameter("print").value
+        self.speak_enabled: bool = self.get_parameter("talk").value
+        self.message_level: int = self.get_parameter("msg_level").value
+
+        # Callback for parameter changes
+        self.add_on_set_parameters_callback(self.on_set_parameters)
+
+        # Mapping from robot name to voice name
+        self.robot_voice_mapping = {
+            "amy": "en_US/vctk_low",
+            "donna": "en_US/vctk_low",
+            "jack": "en_US/vctk_low",
+            "melody": "en_US/vctk_low",
+            "rory": "en_US/vctk_low"
```

Comment:
Do you want to adapt those voices for each robot? I guess, that makes the preloading harder?

Reply:
For now I didn't change the voices and selected the one I liked the most, but we can definitely look into this in the future.

Reply:
I changed them based on gender